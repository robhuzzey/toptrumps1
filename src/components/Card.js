import React from "react";
import Mask from "./Mask";

const Card = ({
  name,
  cost,
  length,
  speed,
  crew,
  passengers,
  capacity,
  hyperdrive,
  reveal
}) => (
  <div>
    <h3>{name}</h3>
    <ul>
      <li>
        Cost: <Mask reveal={reveal}>{cost}</Mask>
      </li>
      <li>
        Length: <Mask reveal={reveal}>{length}</Mask>
      </li>
      <li>
        Speed: <Mask reveal={reveal}>{speed}</Mask>
      </li>
      <li>
        Crew: <Mask reveal={reveal}>{crew}</Mask>
      </li>
      <li>
        Passengers: <Mask reveal={reveal}>{passengers}</Mask>
      </li>
      <li>
        Capacity: <Mask reveal={reveal}>{capacity}</Mask>
      </li>
      <li>
        Hyperdrive: <Mask reveal={reveal}>{hyperdrive}</Mask>
      </li>
    </ul>
  </div>
);

Card.defaultProps = {
  name: "Name",
  costInCredits: 0,
  length: 0,
  speed: 0,
  crew: 0,
  passengers: 0,
  capacity: 0,
  hyperdrive: 0,
  reveal: false,
  winner: false
};

export default Card;
