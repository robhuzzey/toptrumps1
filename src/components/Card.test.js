import React from "react";
import Card from "./Card";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<Card />).toJSON();
  expect(tree).toMatchSnapshot();
});
