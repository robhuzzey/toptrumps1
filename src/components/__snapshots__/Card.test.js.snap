// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h3>
    Name
     (
    
    )
  </h3>
  <ul>
    <li>
      Cost: 
      <span>
        ...
      </span>
    </li>
    <li>
      Length: 
      <span>
        ...
      </span>
    </li>
    <li>
      Speed: 
      <span>
        ...
      </span>
    </li>
    <li>
      Crew: 
      <span>
        ...
      </span>
    </li>
    <li>
      Passengers: 
      <span>
        ...
      </span>
    </li>
    <li>
      Capacity: 
      <span>
        ...
      </span>
    </li>
    <li>
      Hyperdrive: 
      <span>
        ...
      </span>
    </li>
  </ul>
</div>
`;
